util/analyze-gnu-results.py

```diff
@@ -0,0 +1,183 @@
+
+
+def get_priority(result):
+    """Return a priority value for result status (lower is higher priority)"""
+    priorities = {
+        "PASS": 0,  # PASS is highest priority (best result)
+        "FAIL": 1,  # FAIL is second priority
+        "ERROR": 2,  # ERROR is third priority
+        "SKIP": 3,  # SKIP is lowest priority
+    }
+    return priorities.get(result, 4)  # Unknown states have lowest priority
+
+
+def aggregate_results(json_files):
+    """
+    Aggregate test results from multiple JSON files.
+    Prioritizes results in the order: SKIP > ERROR > FAIL > PASS
+    """
+    # Combined results dictionary
+    combined_results = defaultdict(dict)
+
+    # Process each JSON file
+    for json_file in json_files:
+        try:
+            with open(json_file, "r") as f:
+                data = json.load(f)
+
+            # For each utility and its tests
+            for utility, tests in data.items():
+                for test_name, result in tests.items():
+                    # If this test hasn't been seen yet, add it
+                    if test_name not in combined_results[utility]:
+                        combined_results[utility][test_name] = result
+                    else:
+                        # If it has been seen, apply priority rules
+                        current_priority = get_priority(
+                            combined_results[utility][test_name]
+                        )
+                        new_priority = get_priority(result)
+
+                        # Lower priority value means higher precedence
+                        if new_priority < current_priority:
+                            combined_results[utility][test_name] = result
+        except FileNotFoundError:
+            print(f"Warning: File '{json_file}' not found.", file=sys.stderr)
+            continue
+        except json.JSONDecodeError:
+            print(f"Warning: '{json_file}' is not a valid JSON file.", file=sys.stderr)
+            continue
+
+    return combined_results
+
+
+def analyze_test_results(json_data):
+    """
+    Analyze test results from GNU test suite JSON data.
+    Counts PASS, FAIL, SKIP results for all tests.
+    """
+    # Counters for test results
+    total_tests = 0
+    pass_count = 0
+    fail_count = 0
+    skip_count = 0
+    xpass_count = 0  # Not in JSON data but included for compatibility
+    error_count = 0  # Not in JSON data but included for compatibility
+
+    # Analyze each utility's tests
+    for utility, tests in json_data.items():
+        for test_name, result in tests.items():
+            total_tests += 1
+
+            if result == "PASS":
+                pass_count += 1
+            elif result == "FAIL":
+                fail_count += 1
+            elif result == "SKIP":
+                skip_count += 1
+            elif result == "ERROR":
+                error_count += 1
+            elif result == "XPASS":
+                xpass_count += 1
```

Comment:
Since [python 3.10](https://docs.python.org/3/whatsnew/3.10.html) there is also a `match` available:
```suggestion
            match result:
                case "PASS":
                    pass_count += 1
                case "FAIL":
                    fail_count += 1
                case "SKIP":
                    skip_count += 1
                case "ERROR":
                    error_count += 1
                case "XPASS":
                    xpass_count += 1
```